### ISSUE TITLE
Subscribe not sending correct message to device

### ISSUE THREAD

**Issue by skippy1976** [arduino-libraries/ArduinoBLE#204]
I am using an Arduino 33 IOT.  I have a perifpheral that I have captured the services, characteristics and descriptors based on discovery using ArduinoBLE:

```
Service 1800
	Characteristic 2a00, properties 0xA READ WRITE, value 0x4573616E674C696E6B6572
Service fff0
	Characteristic ae01, properties 0x4 WRITTENWITHOUTRESPONSE
		Descriptor 2803, value 0x10080002AE
		Descriptor ae02, value 0x
	Characteristic ae02, properties 0x10 NOTIFY
		Descriptor 2902, value 0x0000
		Descriptor 2803, value 0x040B00F2FF
		Descriptor fff2, value 0x
	Characteristic fff2, properties 0x4 WRITTENWITHOUTRESPONSE
		Descriptor 2803, value 0x100D00F1FF
		Descriptor fff1, value 0x
	Characteristic fff1, properties 0x10 NOTIFY
		Descriptor 2902, value 0x0000
		Descriptor 2803, value 0x20100005AE
		Descriptor ae05, value 0x
	Characteristic ae05, properties 0x20
		Descriptor 2902, value 0x0000
		Descriptor 2803, value 0x0A130010AE
		Descriptor ae10, value 0x3031323334353637
	Characteristic ae10, properties 0xA READ WRITE, value 0x3031323334353637
```

I have also been using the App nRF Connect on my Android device.  The information above matches what nRF provides me, so far so good.

I am trying to subscribe to the charachteristic FFF2 belonging to service FFF0.  in order to receive notificaitons when the value changes.  However, this has not been working.  Using the nRF app I have been able to do this with no issues.  When I turned on BLE Debug, I can see the message being sent to request notifications.  However, this message is different than what nRF is sending.  

Any ideas why these messages are different, and are these differences why the subscribe is not working?  

Arudino message:

```
Subscribing to characteristic fff1
HCI ACLDATA TX -> 020000090005000400120E000100
HCI EVENT RX <- 0413050100000100
HCI ACLDATA RX <- 02002005000100040013
HCI ACLDATA RX <- 02002010000C000500121308001000180000005802
HCI ACLDATA TX -> 0200000A0006000500131302000000
HCI COMMAND TX -> 0113200E0000100018000000580204000600
HCI EVENT RX <- 040F0400051320
```

nRF messages:

```
V	14:46:12.742	Enabling notifications for 0000fff1-0000-1000-8000-00805f9b34fb
D	14:46:12.742	gatt.setCharacteristicNotification(0000fff1-0000-1000-8000-00805f9b34fb, true)
D	14:46:12.743	gatt.writeDescriptor(00002902-0000-1000-8000-00805f9b34fb, value=0x0100)
I	14:46:13.000	Data written to descr. 00002902-0000-1000-8000-00805f9b34fb, value: (0x) 01-00
A	14:46:13.000	"Notifications enabled" sent
V	14:46:13.002	Notifications enabled for 0000fff1-0000-1000-8000-00805f9b34fb
```

**Comment by skippy1976** [comment]
I have looked into the source code of the library and it appears to be matching against the correct descriptor 0x2902, just not sure why the UUID being written is different.  

nRF gives:  `0000290200001000800000805f9b34fb0100`
ArduinoBLE gives: ` 020000090005000400120E000100`

**Comment by skippy1976** [comment]
This maybe related and also could be unreleated, in working through trying to figure out what is going on, I see in BLERemoteCharacteristic.cpp

```
void BLERemoteCharacteristic::setEventHandler(BLECharacteristicEvent event, BLECharacteristicEventHandler eventHandler)
{
  if (event == BLEUpdated) {
    _valueUpdatedEventHandler = eventHandler;
  }
}
```

This implies that you cannot create an Event Handler for when a notification (e.g. BLENotify) for a Characteristic value is updated.  I did change my code to use BLEUpdated but did not get a notification.  

I would suggest changing setEventHandler to return bool based on if it is sucessful or not.  This would mean that if I called setEventHandler with event = BLENotify it would return false.
